plugin/rename.py

```diff
@@ -93,29 +94,26 @@ def run(
         event: Optional[dict] = None,
         point: Optional[int] = None
     ) -> None:
-        if position is None:
-            tmp_pos = get_position(self.view, event, point)
-            if tmp_pos is None:
+        listener = self.get_listener()
+        if listener:
+            listener.purge_changes_async()
```

Comment:
I've added "purge" because I've noticed an issue with lsp-typescript where it requests client to apply edits and immediately triggers rename before client has triggered "didChange".

https://github.com/typescript-language-server/typescript-language-server/blob/0403b36a6ae181fb25f8236f47029afeb2a739fb/src/lsp-server.ts#L938-L949